File: 151003/Romanko/project/publisher/src/main/java/org/example/publisher/impl/sticker/Sticker.java
package org.example.publisher.impl.sticker;

import org.example.publisher.impl.story.Story;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigInteger;
import java.util.List;

@Entity
@Table(name = "tbl_sticker")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Sticker {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    BigInteger id;

    @Column(name = "name")
    private String name;

    @ManyToMany
    @JoinTable(
            name = "tbl_m2m_story_sticker",
            joinColumns = @JoinColumn(name = "sc_id"),
            inverseJoinColumns = @JoinColumn(name = "story_id")
    )
    private List<Story> storys;
}

File: 151004/Akimenko/DC_task_04/kafka/Publisher/Entity/Db/Sticker.cs
using Publisher.Entity.Common;
using System.ComponentModel.DataAnnotations;

namespace Publisher.Entity.Db
{
    public class sticker(string name) : AbstractEntity
    {
        [MinLength(2)]
        public string Name { get; set; } = name;
        public ICollection<story> storys { get; set; } = [];

        public sticker() : this(string.Empty) { }
        public sticker(int id, string name) : this(name)
        {
            Id = id;
        }
    }
}
